### PR TITLE
Add Explanation for Certificate Refreshes

### PR DESCRIPTION
**Pull request by mateoflorido**
## Overview
Add an explanation page that shows certificate refreshes. What they are, their purpose, and how they improve cluster security.

### CODE REVIEW

File: docs/src/snap/explanation/refresh-certs.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+### Preventing Downtime
+
+As previously mentioned, Kubernetes relies on certificates for internal
+communication between components. If these certificates expire and are not
+renewed, critical components like kubelet, API server, or the datastore may
+fail to communicate, leading to operational issues such as cluster downtime
+or disruptions to specific workloads.
+
+Proactively refreshing certificates before they expire ensures that Kubernetes
+maintains continuous and uninterrupted operation of the cluster its workloads.
```
Comment (louiseschmidtgen) [inline docs/src/snap/explanation/refresh-certs.md:51]:
```suggestion
Kubernetes relies on certificates for internal communication between critical components, such as the
kubelet, API server, and datastore. Expired certificates can hinder this communication, leading to potential
downtime and workload disruptions. Proactively refreshing certificates before they expire helps maintain
uninterrupted cluster operations.
```
All your paragraphs are very well written but could be made more concise by removing repetitive information and making the content more dense. Here’s an example, please try and do something similar for the other paragraphs :)

File: docs/src/snap/explanation/refresh-certs.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+### Security Compliance
+
+Security standards, such as [CIS][], often require the regular rotation of
+credentials, including certificates. Periodically renewing Kubernetes
+certificates not only aligns with best practices for security and
+organizational policies but also ensures that the cluster meets security
+standards and compliance requirements.
```
Comment (louiseschmidtgen) [inline docs/src/snap/explanation/refresh-certs.md:59]:
```suggestion
credentials, including certificates. Periodically renewing Kubernetes
certificates ensures that the cluster meets security standards and compliance
requirements.
```

**Review by louiseschmidtgen — COMMENTED** [review]
Thanks for your great work Mateo! Let's try to be a bit more to the point with our explanation in some places.

**Review by louiseschmidtgen — APPROVED** [review]
Wonderful, thanks for addressing my comments!

File: docs/src/snap/explanation/certificates.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
+# Certificates
+
+## What Are Certificates in Kubernetes?
```
Comment (evilnick) [inline docs/src/snap/explanation/certificates.md:3]:
```suggestion

```
Just remove this heading - the follow on paragraph is fine without it

File: docs/src/snap/explanation/certificates.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+a specific validity period, after which they expire and are no longer
+considered valid. Expired certificates lead to failures in communication
+between cluster components, potentially disrupting the entire cluster
+functionality.
+
+Regular certificate refreshes ensure that the certificates in use are valid and
+secure, preventing unauthorized access or communication failures.
+
+## Importance of Certificate Refreshes
+
```
Comment (evilnick) [inline docs/src/snap/explanation/certificates.md:28]: needs text, you can't have stacked headings

File: docs/src/snap/explanation/certificates.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+### Maintaining Cluster Security
+
+Certificates are crucial for securing Kubernetes clusters, ensuring encrypted
+and authenticated communication between components. If a certificate expires
+and isn't promptly renewed, it can leave the cluster vulnerable to security
+risks, potentially exposing it to unauthorized access. Regular certificate
```
Comment (evilnick) [inline docs/src/snap/explanation/certificates.md:34]:
```suggestion
risks, potentially exposing it to unauthorised access. Regular certificate
```

File: docs/src/snap/explanation/certificates.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+API server, kubelets, and the datastore) is both encrypted and restricted to
+authorised components only.
+
+In Kubernetes, [X.509][] certificates are primarily used for
+[Transport Layer Security][] (TLS), securing the communication between the
+cluster's components.
+
+## What Is a Certificate Refresh?
+
+A certificate refresh in {{product}} refers to the process of renewing or
```
Comment (bschimke95) [inline docs/src/snap/explanation/certificates.md:15]: Is there a difference between renewing and rotating certificates?

**Review by bschimke95 — APPROVED** [review]
LGTM, only a minor nit

File: docs/src/snap/explanation/certificates.md (hunk 6 above)
Comment (mateoflorido) [reply docs/src/snap/explanation/certificates.md:15]: We call it refresh (e.g., `refresh-certs`), but the term is interchangeable with renew or rotate.

**Pull request merged**